### PR TITLE
Task/54 sign up and selector function

### PR DESCRIPTION
**Pull request by sggenDebugs**
# Problem:
- sign up has additional inputs
- I want website to let me choose if sign up or login
- not clear if user already exists when registering

# Solution:
- add v-models for the additional inputs
- add function for user in selector page to choose either login or signup
- add pop-up error message in signup to activate when user already exists

### CODE REVIEW

**Review by nonsensenv — DISMISSED** [review]

File: components/Selector/Page.vue [hunk 1]
```diff
@@ -19,6 +19,13 @@
 </template>
 
 <script setup>
```
Comment (Techies-John) [inline components/Selector/Page.vue:21]:
```suggestion
 <script setup lang="ts">
```

File: components/Signup/Page.vue [hunk 2]
```diff
@@ -26,56 +26,57 @@
...
 const router = useRouter();
 const email = ref('');
 const password = ref('');
-// const errorMsg = ref("");
+const name = ref('');
+const idnum = ref('');
+const course = ref('');
+
+let errorMessage = ref('');
 // const successMsg = ref("");
```
Comment (Techies-John) [inline components/Signup/Page.vue:49]: Remove if unused

File: components/login/Page.vue [hunk 3]
```diff
@@ -11,30 +11,16 @@
           <div class="form-group">
             <div class="input-wrapper">
               <label :class="{ active: email.length > 0 || isEmailFocused }" for="email">Silliman Email</label>
```
Comment (Techies-John) [inline components/login/Page.vue:13]: No logic in template use computed

File: components/login/Page.vue [hunk 4]
```diff
@@ -11,30 +11,16 @@
...
-              <input
-                type="email"
-                id="email"
-                v-model="email"
-                class="form-control"
-                @focus="isEmailFocused = true"
-                @blur="isEmailFocused = email.length > 0"
-                required
-              />
+              <input type="email" id="email" v-model="email" class="form-control" @focus="isEmailFocused = true"
```
Comment (Techies-John) [inline components/login/Page.vue:14]:
```suggestion
              <input type="email" id="email" v-model="email" class="form-control" @focus="isEmailFocused"
```

File: components/login/Page.vue [hunk 5]
```diff
@@ -11,30 +11,16 @@
...
+                @blur="isEmailFocused = email.length > 0" required />
             </div>
           </div>
 
           <div class="form-group">
             <div class="input-wrapper">
               <label :class="{ active: password.length > 0 || isPasswordFocused }" for="password">Password</label>
```
Comment (Techies-John) [inline components/login/Page.vue:21]: No logics in template use computed

File: components/login/Page.vue [hunk 6]
```diff
@@ -11,30 +11,16 @@
...
-                :type="passwordInputType"
-                id="password"
-                v-model="password"
-                class="form-control"
-                @focus="isPasswordFocused = true"
-                @blur="isPasswordFocused = password.length > 0"
-                required
-              />
+              <input :type="passwordInputType" id="password" v-model="password" class="form-control"
+                @focus="isPasswordFocused = true" @blur="isPasswordFocused = password.length > 0" required />
```
Comment (Techies-John) [inline components/login/Page.vue:23]:
```suggestion
                @focus="isPasswordFocused" @blur="isPasswordFocused = password.length > 0" required />
```

File: components/login/Page.vue [hunk 7]
```diff
@@ -11,30 +11,16 @@
...
               <div class="eye-icon" v-if="password.length > 0" @click="togglePassword">
```
Comment (Techies-John) [inline components/login/Page.vue:24]: Password.length > 0 is repeating. Can you use computed instead?

File: pages/login.vue [hunk 8]
```diff
@@ -1,11 +1,18 @@
...
 
 <script setup lang="ts">
+// definePageMeta({
+//   middleware: [
+//     function (to, from) {
+//       // Custom inline middleware
+//     },
+//     'welcome',
+//   ],
+// });
```
Comment (Techies-John) [inline pages/login.vue:15]: Remove if unused

File: components/login/Page.vue [hunk 9]
```diff
@@ -11,30 +11,16 @@
...
-                type="email"
-                id="email"
-                v-model="email"
-                class="form-control"
-                @focus="isEmailFocused = true"
-                @blur="isEmailFocused = email.length > 0"
-                required
-              />
+              <input type="email" id="email" v-model="email" class="form-control" @focus="isEmailFocused = true"
+                @blur="isEmailFocused = email.length > 0" required />
```
Comment (Techies-John) [inline components/login/Page.vue:15]: No logics in template use computed

File: components/Signup/Page.vue [hunk 10]
```diff
@@ -26,56 +26,57 @@
...
-const user = useSupabaseUser();
 const router = useRouter();
 const email = ref('');
 const password = ref('');
-// const errorMsg = ref("");
+const name = ref('');
+const idnum = ref('');
+const course = ref('');
+
+let errorMessage = ref('');
```
Comment (Techies-John) [inline components/Signup/Page.vue:48]:
```suggestion
const errorMessage = ref('');
```
Try using const

**Review by Techies-John — CHANGES_REQUESTED** [review]
I will be the one to resolve conversations. Message me if you're done with the changes. Thanks.

File: components/Signup/Page.vue (hunk 10 above)
Comment (sggenDebugs) [reply components/Signup/Page.vue:48]: done

File: components/Signup/Page.vue (hunk 2 above)
Comment (sggenDebugs) [reply components/Signup/Page.vue:49]: done

File: components/Selector/Page.vue (hunk 1 above)
Comment (sggenDebugs) [reply components/Selector/Page.vue:21]: done

File: pages/login.vue (hunk 8 above)
Comment (sggenDebugs) [reply pages/login.vue:15]: done

File: components/login/Page.vue (hunk 9 above)
Comment (sggenDebugs) [reply components/login/Page.vue]: done

File: components/login/Page.vue (hunk 7 above)
Comment (sggenDebugs) [reply components/login/Page.vue:24]: done

**Review by Techies-John — CHANGES_REQUESTED** [review]
I'm not sure what changes were made in nuxt.config.ts, but the login page is currently broken on my machine. After entering credentials, I'm unable to access the `/welcome` or `/` routes. Could you take a look and let me know?

**Comment by Techies-John** [comment]
Inigo, it seems you not have pulled from `main` before working on this task, which could cause issues during merging. Before starting work on a branch, please ensure it's up to date with `main`.

**Review by nonsensenv — APPROVED** [review]

**Review by Techies-John — APPROVED** [review]

**Pull request merged**